bodyshopbidsdotcom/git-scripts
# w278318297 Make Memory and CPU Configurable

Closes https://www.wrike.com/open.htm?id=278318297

## What
Add parser, schema, validation, and test cases for a configuration file that will allow git-scripts to orchestrate deployments and configuration changes from a file saved in a projects root path.

## Why
We used the `scheduled_tasks.yml` and `Procfile` to handle configuration, but both of these files lack the detailed information needed to enable process configuration when deploying. A unified configuration file will enable us to manage more fine-tuned configuration settings.

## How
I am leveraging a package called [`strictyaml`](https://github.com/crdoconnor/strictyaml). This enables us to create schemas for easier validation of configuration format, and manage them as versions.

## Testing
1. Checkout the `add-config-file` branch ([link to the config file](https://github.com/bodyshopbidsdotcom/shops-data/blob/8e5a1b612478d01ba918d1246b39f386d54005f5/infrastructure.yml)) of the `shops-data` project.
1. Confirm it can push to the `stage` environment.
1. Checkout the `develop` branch of the `shops-data` project.
1. Confirm it can push to the `stage` environment.
1. Check out any other repository migrated to AWS.
1. Try to push to one of its environments.
